<a href="https://colab.research.google.com/github/Moez145/day-1-mlintern/blob/main/day-1-mlintern.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# IMPORT LIBRARIES
import numpy as np
import pandas as pd
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.compose import ColumnTransformer
from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import Pipeline
from sklearn.metrics import accuracy_score, classification_report

In [2]:
# LOAD DATASETS
iris = load_iris()
# CONVERT ARRAY TO DATAFRAME
df=pd.DataFrame(iris.data,columns=iris.feature_names)
df['target']=iris.target
df.sample(10)

,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),target
58,6.6,2.9,4.6,1.3,1
133,6.3,2.8,5.1,1.5,2
35,5.0,3.2,1.2,0.2,0
68,6.2,2.2,4.5,1.5,1
145,6.7,3.0,5.2,2.3,2
129,7.2,3.0,5.8,1.6,2
88,5.6,3.0,4.1,1.3,1
31,5.4,3.4,1.5,0.4,0
0,5.1,3.5,1.4,0.2,0
2,4.7,3.2,1.3,0.2,0


In [3]:
# INFO OF THE DATASET
print("THIS IS THE INFO OF THE DATASET.",df.info(),'\n')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
 4   target             150 non-null    int64  
dtypes: float64(4), int64(1)
memory usage: 6.0 KB
THIS IS THE INFO OF THE DATASET. None 



In [4]:
# DESCRIBE OF THE DATASET
df.describe()

,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),target
count,150.000000,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.057333,3.758000,1.199333,1.000000
std,0.828066,0.435866,1.765298,0.762238,0.819232
min,4.300000,2.000000,1.000000,0.100000,0.000000
25%,5.100000,2.800000,1.600000,0.300000,0.000000
50%,5.800000,3.000000,4.350000,1.300000,1.000000
75%,6.400000,3.300000,5.100000,1.800000,2.000000
max,7.900000,4.400000,6.900000,2.500000,2.000000


In [5]:
# IS THERE ANY NULL VALUE IN THE DATASET
df.isnull().sum()

,0
sepal length (cm),0
sepal width (cm),0
petal length (cm),0
petal width (cm),0
target,0


In [6]:
# Train Test Split
x=df.drop('target',axis=1)
y=df['target']

x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.2,random_state=42)
x_train.shape,x_test.shape,y_train.shape,y_test.shape

((120, 4), (30, 4), (120,), (30,))

In [7]:
numerical_columns = x.select_dtypes(include=['int64', 'float64']).columns

# Create Pipeline
numerical_pipeline = Pipeline([
    ('Standard_Scaler', StandardScaler())
])

# Create ColumnTransformer
preprocessor = ColumnTransformer([
    ('numerical', numerical_pipeline, numerical_columns)
])

In [8]:
# KNN model
knn = KNeighborsClassifier(n_neighbors=5)

# Complete pipeline
knn_pipeline = Pipeline([
    ('preprocessor', preprocessor),
    ('model', knn)
])

# Train
knn_pipeline.fit(x_train, y_train)

# Prediction
y_pred = knn_pipeline.predict(x_test)

In [9]:
print("Accuracy:", accuracy_score(y_test, y_pred))
print(classification_report(y_test, y_pred))

Accuracy: 1.0
              precision    recall  f1-score   support

           0       1.00      1.00      1.00        10
           1       1.00      1.00      1.00         9
           2       1.00      1.00      1.00        11

    accuracy                           1.00        30
   macro avg       1.00      1.00      1.00        30
weighted avg       1.00      1.00      1.00        30



In [12]:
# Decision Tree
tree = DecisionTreeClassifier(
    random_state=42
)

# Complete pipeline
tree_pipeline = Pipeline([
    ('preprocessor', preprocessor),
    ('model', tree)
])

# Train
tree_pipeline.fit(x_train, y_train)

# Prediction
y_pred = tree_pipeline.predict(x_test)

print("Accuracy:", accuracy_score(y_test, y_pred))
print(classification_report(y_test, y_pred))

Accuracy: 1.0
              precision    recall  f1-score   support

           0       1.00      1.00      1.00        10
           1       1.00      1.00      1.00         9
           2       1.00      1.00      1.00        11

    accuracy                           1.00        30
   macro avg       1.00      1.00      1.00        30
weighted avg       1.00      1.00      1.00        30



Note:

This is the iris dataset we get it from the build-in dataset from the sklearn moreover it's toy dataset.It contain 150 rows and 5 columns.There is no null value in it.It already well design and convert the text data in numbers(int,float). One important thing is that when we load_iris data it is in the form of array we convert array data in the DataFrame so we able to use pandas.I split data in train_test_split to categories which data is use for training and which one is use for testing.I divide data in 20 , 80 means 20 percent data for testing and 80 percent data for training.I also make pipeline for this problem because it is the professional way to perform a task.This is the simple overview of my working.